contracts/staking/utils/StakingRewardsPerEpoch.sol

```diff
@@ -58,109 +55,85 @@ contract StakingRewardsPerEpoch is ReentrancyGuard, Pausable {
 	}
 
 	function _balanceOf(address account) internal view returns (uint256) {
-		// The resulting balance of any user is the sum of an active earning balance
-		// and pending waiting balance.
-		return stakersInfo[account].balance + stakersInfo[account].pendingStake;
+		return stakersInfo[account].balance;
 	}
 
-	function _getRewardPerTokenPerUser(address _account)
+	function _rewardPerToken()
 		internal
-		view
 		returns (uint256)
 	{
-		// The userEpochIndex is used to calculate a reward per token per epoch
-		// getting into account passed epochs.
-		uint256 userEpochIndex = stakersInfo[_account].indexOfLastEpochStaked + 1;
-		if (lastEpochIndex > userEpochIndex) {
-			// Here we calculate the reward getting into account passed epochs.
-			return rewardsPerTokenAt[lastEpochIndex] - rewardsPerTokenAt[userEpochIndex];
-		} else {
-			return 0;
-		}
+    if (totalSupply == 0) {
+      return rewardPerTokenStored;
+    }
+
+    return
+      rewardPerTokenStored +=
+        (block.timestamp - lastUpdateTime) * interestRatePerBlock;
 	}
 
 	/**
 	 * @dev The function allows anyone to calculate the exact amount of reward
 	 * earned per epochs passed.
 	 * @param account A staker address
 	 */
-	function earned(address account) public view returns (uint256) {
+	function earned(address account) public returns (uint256) {
 		return
-			(stakersInfo[account].balance * _getRewardPerTokenPerUser(account))
+			stakersInfo[account].balance * 
+      (_rewardPerToken() - stakersInfo[account].rewardPerTokenPaid)
 				/ PRECISION
 				+ stakersInfo[account].reward;
 	}
 
-	function _addPendingStakesToBalanceOnEpoch(address _account) internal {
-		// If stakers balance wasn't updated when he staked and the staked
-		// amount is greater than 0, then update active earning balance, nullify
-		// pending one and update global sum of pending stakes.
-		if (stakersInfo[_account].indexOfLastEpochStaked != lastEpochIndex
-				&& stakersInfo[_account].pendingStake > 0) {
-			stakersInfo[_account].balance += stakersInfo[_account].pendingStake;
-      pendingStakes -= stakersInfo[_account].pendingStake;
-			stakersInfo[_account].pendingStake = 0;
-		}
-	}
-
 	// this function updates the reward for the specific user
 	function _updateReward(address _account) internal virtual {
-		_addPendingStakesToBalanceOnEpoch(_account);
-		stakersInfo[_account].reward = earned(_account);
+		    rewardPerTokenStored = _rewardPerToken();
+        lastUpdateTime = block.timestamp;
+        if (_account != address(0)) {
+            stakersInfo[_account].reward = earned(_account);
```

Comment:
stakersInfo[_account].reward += earned(_account);

remove the `+` from earned

Reply:
right, removed